chain/execution_block.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/internal/workers"
+)
+
+type ExecutionBlock struct {
```

Comment:
I'm wondering if this needs to be exported. It seems to me like an execution block represents a block that's being executed, and I don't know why things would need to depend on it outside of the execution (seems like this is what the `chain` package is currently). I wonder if consumers of `chain` should just depend on the `ExecutedBlock` instead?

Reply:
This is used to handle async signature verification and caching the `txsSet` when checking for repeat transactions.

This PR is only meant to re-structure the code, but keep those functions the same and keep a `StatelessBlock` type that has nothing to do with them.